File: SimulationJava/src/Barszcz.java
import javax.swing.ImageIcon;
import javax.swing.JLabel;

public class Barszcz extends Roslina {

    Barszcz(int xx,int yy, Swiat s){
        x=xx;
        y=yy;
        inicjatywa=0;
        sila=10;
        this.s=s;
        if(s instanceof Hexagony) obrazek = new JLabel(new ImageIcon("barszczhex.png"));
        else obrazek = new JLabel(new ImageIcon("barszsosnowskiego.png"));
    }

    @Override
    protected void kolizja(int xx,int yy){
        if(s.getPole(xx, yy).wypiszNazwe() != "cyber-owca"){
            zycie=0;
            makeLog(s.getPole(xx, yy).wypiszNazwe(), wypiszNazwe(), 3, x, y);
            s.getPole(xx, yy).setZycie(martwy);
            s.setTab(x, y, null);
            s.setTab(xx, yy, null);
        } else {
            zycie=0;
            makeLog(s.getPole(xx, yy).wypiszNazwe(), wypiszNazwe(), 2, x, y);
            s.setTab(x, y, s.getPole(xx, yy));
            s.setTab(xx, yy, null);
            s.getPole(x, y).setX(x);
            s.getPole(x, y).setY(y);
        }
    }

    @Override
    public void akcja() {
        int[][] t=s.getSasiadBarszczu(this);

        for(int i=0;i<s.getIleSB();i++){
            if(t[i][xpos]!=-1){
                if(s.getPole(t[i][xpos], t[i][ypos])!=null){
                    if(s.getPole(t[i][xpos], t[i][ypos]).wypiszNazwe()!="cyber-owca" && s.getPole(t[i][xpos], t[i][ypos]).getInicjatywa()>0){
                        if(s.getPole(t[i][xpos], t[i][ypos]).wypiszNazwe()=="czlowiek" && s.getUmiejetnosc()) continue;
                        s.getPole(t[i][xpos], t[i][ypos]).setZycie(0);
                        makeLog(wypiszNazwe(), s.getPole(t[i][xpos], t[i][ypos]).wypiszNazwe(), 5, t[i][xpos], t[i][ypos]);
                        s.setTab(t[i][xpos], t[i][ypos],null);
                    }
                }
            }
        }

       if(probaRozsiania()){
           Organizm n = dziecko();
           rozsiej(n);
       }
    }

    @Override
    public Organizm dziecko() {
        return new Barszcz(x,y,s);
    }

    @Override
    public String wypiszNazwe() {
        return "barszcz sosnowskiego";
    }

    
    @Override
    public JLabel wypisz() {
        return obrazek;
    }
    
}

File: SimulationCpp/projektPO/barszcz.cpp
#include "barszcz.h"
#include "cyberOwca.h"
#include <iostream>
using namespace std;

Barszcz::Barszcz(int x, int y, Swiat* s) {
	sila = 10;
	this->x = x;
	this->y = y;
	zycie = ZYWY;
	this->s = s;
}

bool Barszcz::probaRozsiania() {
	int i = rand() % SZANSANAROZSIANIEBARSZCZU;
	if (i == UDANA)return true;
	return false;
}

void Barszcz::wypisz()const {
	SetConsoleTextAttribute(GetStdHandle(STD_OUTPUT_HANDLE), KOLORBARSZCZU);
	char a = SYMBOLBARSZCZU;
	cout << a;
}

void Barszcz::akcja(){
	for (int i = SASIADGORA; i < SASIADDOL; i++) {
		if (x + i < POCZATEKPLANSZY || x + i >= s->getXM()) continue;
		for (int j = SASIADLEWO; j < SASIADPRAWO; j++) {
			if (y + j < POCZATEKPLANSZY || y + j >= s->getYM()) continue;
			if (s->tab[x + i][y + j] == this) continue;
			if (s->tab[x + i][y + j]) {
				if (s->tab[x + i][y + j]->getInicjatywa()>0 && !dynamic_cast<CyberOwca*>(s->tab[x+i][y+j])) {
					if (s->tab[x + i][y + j]->wypiszNazwe() == "czlowiek" && s->umiejetnosc) break;
					makeLog(wypiszNazwe(), s->tab[x + i][y + j]->wypiszNazwe(), s->log,to_string(x+i),to_string(y+j),5);
					s->tab[x + i][y + j]->zycie = MARTWY;
					s->tab[x + i][y + j] = nullptr;
				}	
			}
		}
	}
	if (probaRozsiania()) {
		Organizm* n = dziecko();
		rozsiej(n);
	}
}


void Barszcz::kolizja(int nX, int nY) {
	if (s->tab[nX][nY]->wypiszNazwe() != "cyber-owca") {
		zycie = MARTWY;
		s->tab[nX][nY]->zycie = MARTWY;
		makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(x), to_string(y), KOMr4);
		s->tab[nX][nY] = nullptr;
		s->tab[x][y] = nullptr;
	}
	else {
		zycie = MARTWY;
		makeLog(s->tab[nX][nY]->wypiszNazwe(), wypiszNazwe(), s->log, to_string(x), to_string(y), KOMr2);
		s->tab[x][y] = s->tab[nX][nY];
		s->tab[nX][nY] = nullptr;
		s->tab[x][y]->setX(x);
		s->tab[x][y]->setY(y);
	}
}

string Barszcz::wypiszNazwe() const {
	return "barszcz sosnowskiego";
}
Organizm* Barszcz::dziecko() {
	return new Barszcz(x, y, s);
}
